yt/yt/server/job_proxy/stderr_writer.cpp

```diff
@@ -142,12 +144,45 @@ void TStderrWriter::DoWrite(const void* buf_, size_t len)
     Tail_->Write(buf, len);
 }
 
-TString TStderrWriter::GetCurrentData() const
+NApi::TPagedLog TStderrWriter::GetCurrentData(const NApi::TPagedLogReq& request) const
 {
     TStringStream stringStream;
     stringStream.Reserve(GetCurrentSize());
-    SaveCurrentDataTo(&stringStream);
-    return stringStream.Str();
+    SaveCurrentDataTo(&stringStream, request.Offset || request.Limit);
+    auto str = stringStream.Str();
+    i64 endOffset = TotalSize_;
+    const i64 currentFirstAbsolutePos = TotalSize_ > static_cast<i64>(str.size()) ? TotalSize_ - str.size() : 0;
```

Comment:
For simplcity, I'd suggest to use, for example, `offset` for absolute indexation and `index` for indexation in the string. `startOffset` seems fine here